drivers/postgres.go

```diff
@@ -863,32 +863,40 @@ func (db *Postgres) GetPrimaryKeyColumnNames(database, table string) (primaryKey
 
 	splitTableString := strings.Split(table, ".")
 
-	if len(splitTableString) == 1 {
+	if len(splitTableString) != 2 {
 		return nil, errors.New("table must be in the format schema.table")
 	}
 
+	schemaName := splitTableString[0]
+	tableName := splitTableString[1]
+
 	if database != db.CurrentDatabase {
 		err = db.SwitchDatabase(database)
 		if err != nil {
 			return nil, err
 		}
-	}
-
-	defer func() {
-		if r := recover(); r != nil {
-			_ = db.SwitchDatabase(db.PreviousDatabase)
-		}
-	}()
 
-	tableName := splitTableString[1]
+		defer func() {
+			if r := recover(); r != nil {
+				_ = db.SwitchDatabase(db.PreviousDatabase)
+			}
+		}()
```

Comment:
Not sure why you would want to catch a panic here? Or did you intent to call the `db.SwitchDatabase(db.PreviousDatabase)` method in case this method returned an error? Or exactly the other way around maybe (only call the `db.SwitchDatabase(db.PreviousDatabase)` method when this method returned without an error 😅)?

Either was I would expect that switching back would only be needed is we actually switched in the first place 😏

Reply:
I admit i find that code a bit weird, i wanted to switch back to the previous database in case of any error, but maybe there's a better way.

Reply:
Ah check... In that case I will make a PR with a suggested way to achieve that goal in a more idiomatic approach. I will remove this change from this PR so you can look at them in isolation.